mempool/reactor.go

```diff
@@ -23,17 +24,27 @@ const (
 )
 
 // MempoolReactor handles mempool tx broadcasting amongst peers.
+// It maintains a map from peer ID to counter, to prevent gossiping txs to the
+// peers you received it from.
 type MempoolReactor struct {
 	p2p.BaseReactor
 	config  *cfg.MempoolConfig
 	Mempool *Mempool
+
+	idMtx   *sync.RWMutex
```

Comment:
does it has to be a pointer?

Reply:
I thought mutexes being pointers is standard? Its also whats done in: https://gobyexample.com/mutexes

Reply:
Huh. You're probably right. We need to examine the current code because I am pretty sure I saw mutexes without pointers.